keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/KeelRepository.kt

```diff
@@ -61,6 +93,13 @@ interface KeelRepository {
       log.debug("Creating $resource")
       storeResource(resource)
       publisher.publishEvent(ResourceCreated(resource, clock))
+
+      // Account for the case where a resource was paused, then deleted (i.e. removed from management), then
```

Comment:
Calling attention to this comment.

Reply:
Wouldn't it be easier to inject the `ApplicationPauser` into the place where we process these events? That way, we don't have to worry about it here, instead we can worry about it when we handle these events.

Reply:
That's a good idea. I hand't thought about that. Will give it a try.